scripts/ci/jobs/diff-dumps.sh

```diff
@@ -107,70 +107,124 @@ create_offline_dump() {
-            exit 1
-        fi
-    done < <(grep -oE '^[0-9]+\.[0-9]+' "$version_file" | sort --reverse --unique --version-sort)
-    if [[ -z "$latest_version" ]]; then
-      >&2 echo "programmer error: latest_version unset..."
-      exit 1
-    fi
-
-    file_to_check="scanner-v4-defs-${latest_version}.zip"
-
-    curl --silent --show-error --fail --max-time 60 --retry 3 -o "$file_to_check" "https://definitions.stackrox.io/v4/offline-bundles/$file_to_check"
-    unzip -l "$file_to_check"
-    zip scanner-vuln-updates.zip "$file_to_check"
-    echo "$file_to_check added to scanner-vuln-updates.zip"
-
-    cmd=()
-    if is_in_PR_context; then
-        cmd+=(echo "Would do")
-    fi
-    "${cmd[@]}" gsutil cp scanner-vuln-updates.zip gs://scanner-support-public/offline/v1/scanner-vuln-updates.zip
+# scanner_v4_create_and_add_bundles() creates versioned offline bundles for V4
+# and add V4 definitions to the latest offline bundle.  It determines the
+# version to use for each release based on the contents of the
+# VULNERABILITY_VERSION file present in each release tag.
+#
+scanner_v4_create_and_add_bundles() {
+    local latest_bundle="${1:-scanner-vuln-updates.zip}"
+
+    local curl="curl --silent --show-error --max-time 60 --retry 3"
+    local release_version_url="https://raw.githubusercontent.com/stackrox/stackrox/master/scanner/updater/version/RELEASE_VERSION"
+    local vuln_version_pattern="https://raw.githubusercontent.com/stackrox/stackrox/%s/scanner/VULNERABILITY_VERSION"
+
+    [ -f scanner-defs.zip ] || die "missing required bundle inclusion: scanner-defs.zip"
+    [ -f k8s-istio.zip ]    || die "missing required bundle inclusion: k8s-istio.zip"
+    [ -f "$latest_bundle" ] || die "missing latest bundle: $latest_bundle"
+
+    info "existing latest bundle:"
+    unzip -l "$latest_bundle"
+
+    info "parsing release versions from $release_version_url"
+
+    # Read the supported releases file, filter lines with release versions,
+    # parse 4.4.* and 4.5.* to print only the X.Y, otherwise print the whole
+    # X.Y.Z, finally merge and store everything together.
+
+    local releases
+    releases=$($curl --fail "$release_version_url" \
+                   | grep '^[0-9]\+\.[0-9]\+\.[0-9]\+$' \
+                   | sed 's/^\(4\.4\|4\.5\)\..*/\1/' \
+                   | sort -V \
+                   | uniq)
+
+    # For each release, build its offline v4 definitions, and track the "latest
+    # definitions" zip.
+
+    local latest_defs
+
+    info "building offline definitions for the following releases:"
+    echo "$releases" | sed 's/^/  - /'
+
+    while read -r release _; do
+        local version
+        case "$release" in
+            4.4)
+                # We don't support 4.4 offline bundles.
+                continue
+                ;;
+            4.5)
+                # Backward compatibility, the "schema version" is the release itself.
+                version="$release"
+                ;;
+            *)
```

Comment:
So we do plan on continuing to update `RELEASE_VERSION`? I thought we weren't going to do so. https://github.com/stackrox/stackrox/blob/master/.github/workflows/scanner-versioned-definitions-update.yaml#L319

Reply:
The file will not be used to determine versioned bundles to build anymore. But, we will continue to use it to list all release tags where V4 is offered.

I might have clarified that elsewhere, but again, the file is not strictly necessary, since we can derive it from other contexts (like the browsing the references in the git repository). But, it is convenient, and I don't want to jump into adopting other methods before I understand how the release tagging is managed (I have ROX-26053 to investigate further and continue from there). Also the change scope and potential blast radius is reduced if I use that file, for now.